nlmod/read/nhi.py

```diff
@@ -0,0 +1,88 @@
+import os
+import requests
+import logging
+import numpy as np
+import rioxarray
+from ..dims.resample import structured_da_to_ds
+
+logger = logging.getLogger(__name__)
+
+
+def download_file(url, pathname, filename=None, overwrite=False, timeout=120):
+    if filename is None:
+        filename = url.split("/")[-1]
+    fname = os.path.join(pathname, filename)
+    if overwrite or not os.path.isfile(fname):
+        logger.info(f"Downloading {filename}")
+        r = requests.get(url, allow_redirects=True, timeout=timeout)
+        with open(fname, "wb") as file:
+            file.write(r.content)
+    return fname
+
+
+def download_buisdrainage(pathname, overwrite=False):
+    url_bas = "https://thredds.data.nhi.nu/thredds/fileServer/opendap/models/nhi3_2/25m"
+
+    # download resistance
+    url = f"{url_bas}/buisdrain_c_ras25/buisdrain_c_ras25.nc"
+    fname_c = download_file(url, pathname, overwrite=overwrite)
+
+    # download drain depth
+    url = f"{url_bas}/buisdrain_d_ras25/buisdrain_d_ras25.nc"
+    fname_d = download_file(url, pathname, overwrite=overwrite)
+
+    return fname_c, fname_d
+
+
+def add_buisdrainage(
+    ds,
+    pathname=None,
+    cond_var="buisdrain_cond",
+    depth_var="buisdrain_depth",
+    cond_method="average",
+    depth_method="mode",
+):
+    if pathname is None:
+        pathname = ds.cachedir
+    # download files if needed
+    fname_c, fname_d = download_buisdrainage(pathname)
+
+    # make sure crs is set of ds
+    if ds.rio.crs is None:
+        ds = ds.rio.write_crs(28992)
```

Comment:
should we use the EPSG definition in nlmod for this?

Reply:
Somehow this does not give the same problems as with maps (I believe, but did not thoroughly check). It might break stuff, if we change this. If other data sources are also in 28992, they do not need to be transformed, but if we change the crs of the model Dataset to our own definition, an unwanted transformation is applied. The buisdrainage data is in WGS84, so you may be right. Let's check this in the future.